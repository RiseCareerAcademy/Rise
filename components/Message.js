import React from "react";
import { View, TouchableOpacity, StyleSheet, Image, Text } from "react-native";

const Message = props => {
  const { navigate } = props.navigation;
  const style = [];
  style.push(styles.message);
  const profileImg =
    "https://cdn.cnn.com/cnnnext/dam/assets/180923222528-tiger-woods-fedex-playoff-exlarge-169.jpg";
  const croppedImg = profileImg + "=s40-c";

  return (
    <TouchableOpacity
      // onPress={() => alert('message')}
      onPress={() => navigate("Conversation")}
      activeOpacity={0.7}
      style={style}
    >
      <View style={styles.topRow}>
        <View style={styles.userCol}>
          <Image
            style={styles.profile}
            source={{
              uri: croppedImg
            }}
          />
          <Text style={styles.user}>{props.fromUser}</Text>
        </View>
        <View style={styles.timeCol}>
          {/* <Text>{props.dateTime}</Text> */}
          <Text style={styles.date}>{"Yesterday"}</Text>
        </View>
      </View>
      <View style={styles.bottomRow}>
        <Text style={styles.messageText}>{props.message}</Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  message: {
    borderWidth: StyleSheet.hairlineWidth,
    borderColor: "rgb(225,225,225)",
    backgroundColor: "rgb(255, 255, 255)",
    marginHorizontal: 15,
    marginBottom: 15,
    padding: 15,
    borderRadius: 10
  },
  bottomRow: {
    flexDirection: "row",
    marginTop: 20
  },
  topRow: {
    flexDirection: "row"
  },
  userCol: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-start",
    alignItems: "center"
  },
  timeCol: {
    flex: 1,
    justifyContent: "center",
    alignItems: "flex-end"
  },
  profile: {
    height: 40,
    width: 40,
    borderRadius: 20
  },
  user: {
    fontSize: 20,
    marginLeft: 10
  },
  date: {
    color: "rgb(102, 102, 102)"
  },
  messageText: {
    fontSize: 14
  }
});

export default Message;
