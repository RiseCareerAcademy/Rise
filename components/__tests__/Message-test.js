import 'react-native';
import React from 'react';
import Message from '../Message';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const props = {
        message: 'MyMessage',
        fromUser: 'FromUser',
        navigation: {
            navigate: () => {}
        },
    };
  const tree = renderer.create(<Message {...props}>Snapshot test!</Message>).toJSON();

  expect(tree).toMatchSnapshot();
});
