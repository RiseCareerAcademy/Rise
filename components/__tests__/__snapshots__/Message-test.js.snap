// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgb(255, 255, 255)",
      "borderColor": "rgb(225,225,225)",
      "borderRadius": 10,
      "borderWidth": 0.5,
      "marginBottom": 15,
      "marginHorizontal": 15,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cdn.cnn.com/cnnnext/dam/assets/180923222528-tiger-woods-fedex-playoff-exlarge-169.jpg=s40-c",
          }
        }
        style={
          Object {
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
            "marginLeft": 10,
          }
        }
      >
        FromUser
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(102, 102, 102)",
          }
        }
      >
        Yesterday
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 14,
        }
      }
    >
      MyMessage
    </Text>
  </View>
</View>
`;
